stdlib/CompilerSupportLibraries_jll/src/CompilerSupportLibraries_jll.jl

```diff
@@ -5,80 +5,108 @@
+    const _libquadmath_path = BundledLazyLibraryPath("libquadmath.0.dylib")
     if arch(HostPlatform()) == "aarch64" || libgfortran_version(HostPlatform()) == v"5"
-        const libgcc_s = "@rpath/libgcc_s.1.1.dylib"
+        const _libgcc_s_path = BundledLazyLibraryPath("libgcc_s.1.1.dylib")
     else
-        const libgcc_s = "@rpath/libgcc_s.1.dylib"
+        const _libgcc_s_path = BundledLazyLibraryPath("libgcc_s.1.dylib")
     end
-    const libgfortran = string("@rpath/", "libgfortran.", libgfortran_version(HostPlatform()).major, ".dylib")
-    const libstdcxx = "@rpath/libstdc++.6.dylib"
-    const libgomp = "@rpath/libgomp.1.dylib"
-    const libssp = "@rpath/libssp.0.dylib"
+    const _libgfortran_path = BundledLazyLibraryPath(string("libgfortran.", libgfortran_version(HostPlatform()).major, ".dylib"))
+    const _libstdcxx_path = BundledLazyLibraryPath("libstdc++.6.dylib")
+    const _libgomp_path = BundledLazyLibraryPath("libgomp.1.dylib")
+    const _libssp_path = BundledLazyLibraryPath("libssp.0.dylib")
 else
-    const libgcc_s = "libgcc_s.so.1"
-    const libgfortran = string("libgfortran.so.", libgfortran_version(HostPlatform()).major)
-    const libstdcxx = "libstdc++.so.6"
-    const libgomp = "libgomp.so.1"
+    if !Sys.isfreebsd()
+        const _libatomic_path = BundledLazyLibraryPath("libatomic.so.1")
+    end
+    const _libgcc_s_path = BundledLazyLibraryPath("libgcc_s.so.1")
+    const _libgfortran_path = BundledLazyLibraryPath(string("libgfortran.so.", libgfortran_version(HostPlatform()).major))
+    const _libstdcxx_path = BundledLazyLibraryPath("libstdc++.so.6")
+    const _libgomp_path = BundledLazyLibraryPath("libgomp.so.1")
     if libc(HostPlatform()) != "musl"
-        const libssp = "libssp.so.0"
+        const _libssp_path = BundledLazyLibraryPath("libssp.so.0")
+    end
+    if arch(HostPlatform()) ∈ ("x86_64", "i686")
+        const _libquadmath_path = BundledLazyLibraryPath("libquadmath.so.0")
     end
 end
 
+if @isdefined(_libatomic_path)
+    const libatomic = LazyLibrary(_libatomic_path)
+end
+const libgcc_s = LazyLibrary(_libgcc_s_path)
+libgfortran_deps = [libgcc_s]
+if @isdefined _libquadmath_path
+    const libquadmath = LazyLibrary(_libquadmath_path)
+    push!(libgfortran_deps, libquadmath)
+end
+const libgfortran = LazyLibrary(_libgfortran_path, dependencies=libgfortran_deps)
+const libstdcxx = LazyLibrary(_libstdcxx_path, dependencies=[libgcc_s])
+const libgomp = LazyLibrary(_libgomp_path)
+if @isdefined _libssp_path
+    const libssp = LazyLibrary(_libssp_path)
+end
+
+# Conform to LazyJLLWrappers API
+function eager_mode()
+    if @isdefined(libatomic)
+        dlopen(libatomic)
+    end
+    dlopen(libgcc_s)
+    dlopen(libgomp)
+    if @isdefined libquadmath
+        dlopen(libquadmath)
+    end
+    if @isdefined libssp
+        dlopen(libssp)
+    end
+    dlopen(libgfortran)
+    dlopen(libstdcxx)
+end
+is_available() = true
```

Comment:
Out of ignorance, what is this check actually used for? Is it OK that this "lies" if the libraries aren't around to be lazily loaded?

Reply:
This is used to determine if a JLL has a build at all, not that all possible libraries are present. For stdlib JLLs they are always true by construction, except for cases like libunwind perhaps?